lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
Sorry for the slow response, I am concerned both with pythonic implementations, speed and obfuscation combined.

Creating a function named the same as the original method, by which it is hard to tell in the calling location which is happening will often lead to confusion.

hasattr can also sometimes work in strange ways such as what is described in https://hynek.me/articles/hasattr/

Additionally, with hasattr you are performing the lookup multiple times, instead of just trying to use it and handle the exception.  In the majority of cases, just trying to use it and handling the exception will be the faster way.